src/sidecar/sidecarManager.ts

```diff
@@ -417,31 +412,44 @@ export class SidecarManager {
         // Pause after spawning (so as to let the sidecar initialize and bind to its port),
         // then try to hit the handshake endpoint. It may fail a few times while
         // the sidecar process is coming online.
-        for (let i = 0; i < 10; i++) {
+        for (let i = 0; i < max_handshake_attempts; i++) {
           try {
             await this.pause();
 
+            accessToken = await this.doHandshake();
+            await getStorageManager().setSecret(SecretStorageKeys.SIDECAR_AUTH_TOKEN, accessToken);
```

Comment:
Pulled these success path actions up from after the loop. Reads much better now.